src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long productId) {
+        return Optional.ofNullable(Storage.products.stream()
+                .filter(product -> product.getId() == productId)
+                .findFirst().orElseThrow());
```

Comment:
```suggestion
        return Storage.products.stream()
                .filter(product -> product.getId() == productId)
                .findFirst();
```

also, do we use `==` to compare object types?